From 5d10ac5ec6b5f194a6591690ca2ee8bc0522a7cc Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Wed, 15 Feb 2017 16:20:56 -0800
Subject: [PATCH] DAPI-337 dapi's hard-filter-volumes-from mutates
 vm.internal_metadata['volumesfrom'] such that multiple alloc passes from
 CNAPI-671 breaks

---
 lib/algorithms/hard-filter-volumes-from.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/algorithms/hard-filter-volumes-from.js b/lib/algorithms/hard-filter-volumes-from.js
index 203a2ca..5966807 100644
--- a/lib/algorithms/hard-filter-volumes-from.js
+++ b/lib/algorithms/hard-filter-volumes-from.js
@@ -112,13 +112,14 @@ loadVmAndFilter(log, servers, requiredVms, getVm, cb)
 	log.debug('opts.getVm set; fetching VMs listed in volumes-from');
 
 	var reasons = {};
+	var remainingVms = requiredVms.slice(); // Get a copy, b/c we mutate it.
 
 	function step() {
-		if (requiredVms.length === 0 || servers.length === 0) {
+		if (remainingVms.length === 0 || servers.length === 0) {
 			return (cb(null, servers.slice(0, 1), reasons));
 		}
 
-		var vmUuid = requiredVms.shift();
+		var vmUuid = remainingVms.shift();
 
 		return getVm(vmUuid, function (err, vm) {
 			if (err) {
-- 
2.21.0

